### PR TITLE
Added MatMul tests for QNN EP

### PR DESCRIPTION
**Pull request by aungthetnaing**
### Description

Added test coverage for QNN EP MatMul op

### Motivation and Context

Created test coverage for HTP based MatMul with broadcasting.

### CODE REVIEW

File: onnxruntime/test/providers/qnn/matmul_test.cpp [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    builder.AddNode("MatMul", {input1, input2}, {output});
+  };
+}
+
+// Returns a function that creates a graph with a QDQ AveragePool operator.
+template <typename QuantType>
+GetQDQTestCaseFn BuildMatMulOpQDQTestCase(const std::vector<int64_t>& input1_shape,
+                                          const std::vector<int64_t>& input2_shape) {
+  return [input1_shape, input2_shape](ModelTestBuilder& builder) {
+    //float dq_scale = 0.0038f;
```
Comment (HectorSVC) [inline onnxruntime/test/providers/qnn/matmul_test.cpp:37]: >//float dq_scale = 0.0038f; [](start = 4, length = 27)

remove

File: onnxruntime/test/providers/qnn/matmul_test.cpp [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    float pool_output_scale = 0.0038f;
+    float q_scale = 0.0038f;
+    // QuantType dq_zp = std::numeric_limits<QuantType>::max() / 2;
```
Comment (HectorSVC) [inline onnxruntime/test/providers/qnn/matmul_test.cpp:40]: >// QuantType dq_zp = std::numeric_limits<QuantType>::max() / 2; [](start = 4, length = 63)

remove

File: onnxruntime/test/providers/qnn/matmul_test.cpp [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    float pool_output_scale = 0.0038f;
+    float q_scale = 0.0038f;
+    QuantType pool_output_zp = std::numeric_limits<QuantType>::max() / 2;
+    QuantType q_zp = std::numeric_limits<QuantType>::max() / 2;
+
+    auto* input_arg = builder.MakeInput<float>(input1_shape, -1.f, 1.f);
+    auto* output_arg = builder.MakeOutput();
+
+    using InputLimits = std::numeric_limits<QuantType>;
+    using OutputTypeLimits = std::numeric_limits<QuantType>;
```
Comment (HectorSVC) [inline onnxruntime/test/providers/qnn/matmul_test.cpp:46]: >using OutputTypeLimits = std::numeric_limits<QuantType>; [](start = 4, length = 56)

remove

Comment (HectorSVC) [inline onnxruntime/test/providers/qnn/matmul_test.cpp:46]:
```suggestion
```

File: onnxruntime/test/providers/qnn/matmul_test.cpp [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+// QNN broadcast issue
+TEST_F(QnnHTPBackendTests, DISABLED_TestMatMulOp2_HTP_u8) {
+  RunQDQMatMulOpOpTest<uint8_t>({28, 1, 64} /* input_shape1 */,
+                                {64, 32} /* input_shape2 */,
+                                ExpectedEPNodeAssignment::All, "TestMatMulOp2_HTP_u8",
+                                18, 0.00381f);
+}
+
+TEST_F(QnnHTPBackendTests, TestMatMulOp3_HTP_u8) {
```
Comment (HectorSVC) [inline onnxruntime/test/providers/qnn/matmul_test.cpp:167]:
```suggestion
// QNN broadcast issue
TEST_F(QnnHTPBackendTests, DISABLED_TestMatMulOp3_HTP_u8) {
```

**Comment by HectorSVC** [comment]
Could you merge the latest main branch?

**Review by HectorSVC — APPROVED** [review]

**Review by adrianlizarraga — APPROVED** [review]

**Pull request merged**